### PR TITLE
CSI-3242: HyperSwap feature updates

### PR DESCRIPTION
**Pull request by rivkap**

### CODE REVIEW

File: docs/content/release_notes/csi_rn_limitations.md [hunk 1]
```diff
@@ -41,4 +47,14 @@ When a role switch is conducted, this is not reflected within the other orchestr
...
+
+## HyperSwap usage limitations
+
+**Important:** The HyperSwap feature is only supported for use with IBM Spectrum Virtualize Family storage systems.
+
+The following IBM block storage CSI driver features are not supported on the volumes where HyperSwap is being used:
+
+- Volume snapshots
+- Volume cloning
+- Volume expansion
```
Comment (oriyarde) [inline docs/content/release_notes/csi_rn_limitations.md:60]: update: we are working on allowing snapshots and expansion with hyperswap for 1.8.0.
so only cloning won't be supported with hyperswap

File: docs/content/release_notes/csi_rn_whatsnew.md [hunk 2]
```diff
@@ -1,22 +1,13 @@
-# What's new in 1.7.0
+# What's new in 1.8.0
 
-IBM® block storage CSI driver 1.7.0 introduces the enhancements that are detailed in the following section.
+IBM® block storage CSI driver 1.8.0 introduces the enhancements that are detailed in the following section.
 
-**General availability date:** 30 September 2021
+**General availability date:** 30 December 2021
```
Comment (oriyarde) [inline docs/content/release_notes/csi_rn_whatsnew.md:5]: please verify this date with Eran

File: docs/content/configuration/csi_ug_config_create_storageclasses.md [hunk 3]
```diff
@@ -24,14 +24,22 @@ _<a name=spaceefficiency>**Table:**</a> `SpaceEfficiency` parameter definitions
 
 - The IBM DS8000 Family `pool` value is the pool ID and not the pool name as is used in other storage systems.
 - Be sure that the `pool` value is the name of an existing pool on the storage system.
+- To create a volume with HyperSwap on IBM Spectrum Virtualize storage systems, put a colon (:) between the two pools within the `pool` value. For example:
```
Comment (oriyarde) [inline docs/content/configuration/csi_ug_config_create_storageclasses.md:27]: @roysahar / @barakda are there any hyperswap prerequisites to perform on the storage side first?

File: docs/content/release_notes/csi_rn_whatsnew.md (hunk 2 above)
Comment (rivkap) [reply docs/content/release_notes/csi_rn_whatsnew.md:5]: the final date will be changed later and in a later PR. thanks

File: docs/content/release_notes/csi_rn_limitations.md (hunk 1 above)
Comment (rivkap) [reply docs/content/release_notes/csi_rn_limitations.md:60]: thanks. if this is all that is left for this PR (once [this comment](https://github.com/IBM/ibm-block-csi-driver/pull/397#discussion_r742764487) is resolved) i can open another ticket and PR for the limitation changes, if needed.
or, if you prefer, I can remove those two limitations and then just add them back in later if needed.

Comment (rivkap) [reply docs/content/release_notes/csi_rn_limitations.md:60]: @oriyarde  reminder. thanks!

File: docs/content/configuration/csi_ug_config_create_storageclasses.md (hunk 3 above)
Comment (rivkap) [reply docs/content/configuration/csi_ug_config_create_storageclasses.md:27]: @barakda  reminder :) thanks

File: docs/content/release_notes/csi_rn_limitations.md (hunk 1 above)
Comment (oriyarde) [reply docs/content/release_notes/csi_rn_limitations.md:60]: let's remove those (non-cloning) limitations for now

Comment (rivkap) [reply docs/content/release_notes/csi_rn_limitations.md]: DONE

File: docs/content/configuration/csi_ug_config_create_storageclasses.md (hunk 3 above)
Comment (rivkap) [reply docs/content/configuration/csi_ug_config_create_storageclasses.md:27]: Thanks, @roysahar  I've added additional information here as well as in the compatibility file.
Please view and approve.

Comment (oriyarde) [reply docs/content/configuration/csi_ug_config_create_storageclasses.md:27]: >For more information about the HyperSwap limitations

why specifically limitations? this page does not mention anything related specifically to the limitations.
I'm not even sure it is necessary to refer to the limitations from this page, especially not before referring to the requirements

File: docs/content/configuration/csi_ug_config_create_storageclasses.md [hunk 4]
```diff
@@ -24,14 +24,23 @@ _<a name=spaceefficiency>**Table:**</a> `SpaceEfficiency` parameter definitions
...
+  
+  ```
+  pool: demo-pool1:demo-pool2 
+  ```
+   **Important:** The two pools must be from different sites.
+   
+  For more information about the HyperSwap limitations, see [Limitations](content\release_notes\csi_rn_limitations.md) within the release notes and [Compatibility and requirements](content\installation\csi_ug_requirements.md).
```
Comment (oriyarde) [inline docs/content/configuration/csi_ug_config_create_storageclasses.md:34]: any idea why these links lead to 404 not found error?

Comment (rivkap) [reply docs/content/configuration/csi_ug_config_create_storageclasses.md]: fixed. it was a linking issue.
DONE

Comment (oriyarde) [reply docs/content/configuration/csi_ug_config_create_storageclasses.md]: I still get [404](https://github.com/IBM/ibm-block-csi-driver/blob/3723e83c1e96c88cc26c7902f332beccb77abac3/docs/content/configuration/..%5Cinstallation%5Ccsi_ug_requirements.md).
any luck with the link scanner integration?

File: docs/content/release_notes/csi_rn_limitations.md [hunk 5]
```diff
@@ -41,4 +43,10 @@ When a role switch is conducted, this is not reflected within the other orchestr
...
       
-    For more information, see the [Change the Reclaim Policy of a PersistentVolume](https://kubernetes.io/docs/tasks/administer-cluster/change-pv-reclaim-policy/) information in the Kubernetes documentation.
\ No newline at end of file
+    For more information, see the [Change the Reclaim Policy of a PersistentVolume](https://kubernetes.io/docs/tasks/administer-cluster/change-pv-reclaim-policy/) information in the Kubernetes documentation.
+
+## HyperSwap usage limitations
+
+**Important:** The HyperSwap feature is only supported for use with IBM Spectrum Virtualize Family storage systems.
+
+Volume cloning does not work on volumes using the HyperSwap feature.
```
[the remark this answers is not in the document]
Comment (rivkap) [reply docs/content/release_notes/csi_rn_limitations.md]: thanks. fixed.
DONE

[the remark this answers is not in the document]
Comment (oriyarde) [reply docs/content/release_notes/csi_rn_limitations.md]: note that nuance - a hyperswap volume can be snapshotted, but a hyperswap volume cannot be created from a snapshot

File: docs/content/release_notes/csi_rn_limitations.md [hunk 6]
```diff
@@ -41,4 +45,15 @@ When a role switch is conducted, this is not reflected within the other orchestr
...
       
-    For more information, see the [Change the Reclaim Policy of a PersistentVolume](https://kubernetes.io/docs/tasks/administer-cluster/change-pv-reclaim-policy/) information in the Kubernetes documentation.
\ No newline at end of file
+    For more information, see the [Change the Reclaim Policy of a PersistentVolume](https://kubernetes.io/docs/tasks/administer-cluster/change-pv-reclaim-policy/) information in the Kubernetes documentation.
+
+## HyperSwap usage limitations
+
+**Important:** The HyperSwap feature is only supported for use with IBM Spectrum Virtualize Family storage systems.
+
+The following IBM block storage CSI driver features are not supported on the volumes where HyperSwap is being used:
```
Comment (oriyarde) [inline docs/content/release_notes/csi_rn_limitations.md:54]:
```suggestion
The following IBM block storage CSI driver features are not supported on volumes where HyperSwap is being used:
```

File: docs/content/configuration/csi_ug_config_create_storageclasses.md (hunk 3 above)
Comment (rivkap) [reply docs/content/configuration/csi_ug_config_create_storageclasses.md:27]: @oriyarde  I'm assuming you're talking about here: configuration/csi_ug_config_create_storageclasses.md

I've updated the wording a little bit so it's clearer but it's pointing specifically to the limitations page and to the compatibility and requirements pages for more information.
DONE

File: docs/content/configuration/csi_ug_config_create_storageclasses.md (hunk 4 above)
Comment (rivkap) [reply docs/content/configuration/csi_ug_config_create_storageclasses.md]: no luck with the integration but my slashes were backwards. i checked it in the files themselves (not just locally) and it works now

Comment (rivkap) [reply docs/content/configuration/csi_ug_config_create_storageclasses.md]: DONE

File: docs/content/release_notes/csi_rn_limitations.md (hunk 6 above)
Comment (rivkap) [reply docs/content/release_notes/csi_rn_limitations.md]: DONE

**Review by oriyarde — APPROVED** [review]
LGTM

**Pull request merged**